discojs/src/weights/aggregation.spec.ts

```diff
@@ -4,33 +4,68 @@ import { WeightsContainer, aggregation } from './index.js'
 
 describe('weights aggregation', () => {
   it('avg of weights with two operands', () => {
-    const actual = aggregation.avg([
+    const expected = aggregation.avg([
       WeightsContainer.of([1, 2, 3, -1], [-5, 6]),
       WeightsContainer.of([2, 3, 7, 1], [-10, 5]),
       WeightsContainer.of([3, 1, 5, 3], [-15, 19])
     ])
-    const expected = WeightsContainer.of([2, 2, 5, 1], [-10, 10])
+    const actual = WeightsContainer.of([2, 2, 5, 1], [-10, 10])
 
-    assert.isTrue(actual.equals(expected))
+    assert.isTrue(expected.equals(actual))
   })
 
   it('sum of weights with two operands', () => {
-    const actual = aggregation.sum([
+    const expected = aggregation.sum([
       [[3, -4], [9]],
       [[2, 13], [0]]
     ])
-    const expected = WeightsContainer.of([5, 9], [9])
+    const actual = WeightsContainer.of([5, 9], [9])
 
-    assert.isTrue(actual.equals(expected))
+    assert.isTrue(expected.equals(actual))
   })
 
   it('diff of weights with two operands', () => {
-    const actual = aggregation.diff([
+    const expected = aggregation.diff([
       [[3, -4, 5], [9, 1]],
       [[2, 13, 4], [0, 1]]
     ])
-    const expected = WeightsContainer.of([1, -17, 1], [9, 0])
+    const actual = WeightsContainer.of([1, -17, 1], [9, 0])
 
-    assert.isTrue(actual.equals(expected))
+    assert.isTrue(expected.equals(actual))
   })
+
+  it('avg of weights with no operands throws an error', () => {
+    assert.throws(() => aggregation.avg([]))
+  })
+
+  it('sum of weights with no operands throws an error', () => {
+    assert.throws(() => aggregation.sum([]))
+  })
+
+  it('diff of weights with no operands throws an error', () => {
+    assert.throws(() => aggregation.diff([]))
+  })
+
+  it('avg of weights with different dimensions throws an error', () => {
+    assert.throws(() => aggregation.avg([
+      [[3, -4], [9]],
+      [[2, 13, 4], [0, 1]]
+    ]))
+  })
+
+  it('sum of weights with different dimensions throws an error', () => {
+    assert.throws(() => aggregation.sum([
+      [[3, -4], [9]],
+      [[2, 13, 4], [0, 1]]
+    ]))
+  })
+
+  it('diff of weights with different dimensions throws an error', () => {
+    assert.throws(() => aggregation.diff([
+      [[3, -4], [9]],
+      [[2, 13, 4], [0, 1]]
+    ]))
+  })
```

Comment:
this is a bench test, you can simplify it by using a `Set(…).forEach((op) => …)`

while at it, should it fail? an addition without value can be zero, a multiplication can be one; why is this test needed?

```suggestion
  Set.of(aggregation.avg, aggregation.sum, aggregation.diff).forEach((op) => {
    it(`${op.name} of weights with no operands throws`, () =>
      expect(() => op([])).to.throw());

    it(`${op.name} of weights with different dimensions throws`, () =>
      expect(() => op([[[1, 2]], [[3, 4, 5]]])).to.throw());
  });
```